internal/sql/versioning/versioning.go

```diff
@@ -0,0 +1,107 @@
+// package versioning provides standard schema versioning for Kwil databases.
+package versioning
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kwilteam/kwil-db/common/sql"
+)
+
+var (
+	ErrTargetVersionTooLow = fmt.Errorf("target version is lower than current version")
+)
+
+// EnsureVersionTableExists ensures that the version table exists in the database.
+// If the table does not exist, it will be created, and the first version will be set to version specified.
+func ensureVersionTableExists(ctx context.Context, db sql.DB, schema string, targetVersion int64) error {
+	tx, err := db.BeginTx(ctx)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback(ctx)
+
+	// Create the schema if it does not exist
+	if _, err := tx.Execute(ctx, fmt.Sprintf(sqlCreateSchema, schema)); err != nil {
+		return err
+	}
+
+	// Create the version table if it does not exist
+	if _, err = tx.Execute(ctx, fmt.Sprintf(sqlVersionTable, schema)); err != nil {
+		return err
+	}
+
+	// Ensure that the version exists. If it does not, insert it with the target version.
+	if _, err = tx.Execute(ctx, fmt.Sprintf(sqlEnsureVersionExists, schema), targetVersion); err != nil {
```

Comment:
this is almost certainly a bug. If have a new node + new DB and are at version 1, we will essentially be setting the "current version" to version 1. This will then inform subsequent code that we are already at the correct version, when in fact we are at no version